Fix errors in comparison functions when dtype is uint8

YIWENX14 opened the pull request:
Summary: Fixed error in comparison functions that were producing wrong results with uint8 dtype.

Differential Revision: D67163600

YIWENX14 commented:
@pytorchbot label "topic: not user facing"

Olivia-liu reviewed (approved):
Thanks for making the fix!

YIWENX14 committed:
Fix errors in comparison functions when dtype is uint8 (#7313)

Summary:

Fixed error in comparison functions that were producing wrong results with uint8 dtype.

Reviewed By: Olivia-liu

Differential Revision: D67163600
